modules/light-clients/07-tendermint/consensus_host.go

```diff
@@ -0,0 +1,125 @@
+package tendermint
+
+import (
+	"reflect"
+
+	errorsmod "cosmossdk.io/errors"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cometbft/cometbft/light"
+
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v8/modules/core/23-commitment/types"
+	ibcerrors "github.com/cosmos/ibc-go/v8/modules/core/errors"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+)
+
+var _ clienttypes.ConsensusHost = (*ConsensusHost)(nil)
+
+// ConsensusHost implements the 02-client clienttypes.ConsensusHost interface
+type ConsensusHost struct {
+	stakingKeeper clienttypes.StakingKeeper
```

Comment:
todo: this expected keeper interface for `StakingKeeper` should be moved to 07-tendermint as well

Reply:
redefined in 07-tendermint, should be removed from clienttypes if and when the `NewKeeper` arg is changed.

see: https://github.com/cosmos/ibc-go/pull/6055#discussion_r1538903188